cluster/node.go

```diff
@@ -0,0 +1,72 @@
+package cluster
+
+import (
+	"sync"
+	"time"
+)
+
+type NodeState int
+
+const (
+	NodeReady       NodeState = 0
+	NodeNotReady    NodeState = 1
+	NodeUnreachable NodeState = 2
+)
+
+var NodeStateText = map[NodeState]string{
+	NodeReady:       "Ready",
+	NodeNotReady:    "NotReady",
+	NodeUnreachable: "Unreachable",
+}
+
+type Node struct {
+	Name    string    `json:"name"`
+	Version string    `json:"version"`
+	Primary bool      `json:"primary"`
+	State   NodeState `json:"state"`
+	Started time.Time `json:"started"`
+	sync.RWMutex
```

Comment:
is the only reason for `type nodeJS` existence the fact that now the members are un-exported?  if so I think that's a good enough reason to make the members exported again and get rid of nodeJS type. this one of those annoying things about go.. we want to prevent unsafe direct member access but at the same time we want to be able to export the members. I think it's ok to make the members Public so that we don't need this workaround. But up to you i guess.